### DISCUSSION TITLE
Where Can I Install CNI Plugins if I'm not Root?

### DISCUSSION THREAD

**Discussion by zicklag** [containers/podman#12148] [Q&A] [answered]
Hey there!

I'm getting a warning when running podman:

```
WARN[0000] Error validating CNI config file /home/zicklag/.config/cni/net.d/87-podman.conflist: [failed to find plugin "bridge" in path [/usr/local/libexec/cni /usr/libexec/cni /usr/local/lib/cni /usr/lib/cni /opt/cni/bin] failed to find plugin "portmap" in path [/usr/local/libexec/cni /usr/libexec/cni /usr/local/lib/cni /usr/lib/cni /opt/cni/bin] failed to find plugin "firewall" in path [/usr/local/libexec/cni /usr/libexec/cni /usr/local/lib/cni /usr/lib/cni /opt/cni/bin] failed to find plugin "tuning" in path [/usr/local/libexec/cni /usr/libexec/cni /usr/local/lib/cni /usr/lib/cni /opt/cni/bin]]
```

Since I'm not root on my machine, I can't install the CNI plugins to any of those directories. Is there a place that I can put them or an environment variable I can set to tell podman where to look for CNI plugins, so that I can run podman without being root?

**Comment by Luap99** [comment, marked as answer]
You can use [`cni_plugin_dirs` in containers.conf](https://github.com/containers/common/blob/main/docs/containers.conf.5.md#network-table) to change the directories.

**Reply by zicklag** [reply]
Sweet, thanks, it worked!
